src/module.c

```diff
@@ -100,30 +100,44 @@ void ReplyWithAggValue(RedisModuleCtx *ctx, timestamp_t last_agg_timestamp, Aggr
 
 int TSDB_range(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     RedisModule_AutoMemory(ctx);
-    
-    if (argc < 4 || argc > 6) return RedisModule_WrongArity(ctx);
 
     long long start_ts, end_ts;
+    long long time_delta = 0;
+    RedisModuleString * aggTypeStr = NULL;
+
+    int pRes = REDISMODULE_ERR;
+    switch (argc) {
+        case 4:
+            pRes = RMUtil_ParseArgs(argv, argc, 2, "ll", &start_ts, &end_ts);
+            break;
+        /* case 5: return RedisModule_ReplyWithLongLong(ctx, 12341234); */ // Test false positive
+        case 6:
+            pRes = RMUtil_ParseArgs(argv, argc, 2, "llsl", &start_ts, &end_ts, &aggTypeStr, &time_delta );
+            break;
+        default:
+            return RedisModule_WrongArity(ctx);
+    }
+    if (pRes != REDISMODULE_OK)
+        return RedisModule_WrongArity(ctx);
+
     long long agg_type = 0;
-    long long dt = 0;
     Series *series;
     RedisModuleKey *key;
-    AggregationClass *aggObject;
-
-    if (RedisModule_StringToLongLong(argv[2], &start_ts) != REDISMODULE_OK)
-        return RedisModule_ReplyWithError(ctx, "TSDB: start-timestamp is invalid");
-    if (RedisModule_StringToLongLong(argv[3], &end_ts) != REDISMODULE_OK)
-        return RedisModule_ReplyWithError(ctx, "TSDB: end-timestamp is invalid");
-    if (argc > 4) {
-        agg_type = StringAggTypeToEnum(argv[4]);
-        int aggType = StringAggTypeToEnum(argv[4]);
-        if (aggType < 0 && aggType > 5) {
+    AggregationClass *aggObject = NULL;
+
+    if (argc > 4)
+    {
+        if (!aggTypeStr)
+            return RedisModule_ReplyWithError(ctx, "TSDB: Unkown aggregation type");
+        agg_type = StringAggTypeToEnum( aggTypeStr );
+        // int aggType = StringAggTypeToEnum( aggTypeStr );
```

Comment:
@someburner i think this line can be safely deleted, any reason you kept this?

Reply:
I just wasn't sure why there was 2 calls to `StringAggTypeToEnum` in the first place, so wanted to leave to make sure there isn't something I was missing. So if not, then yes we can delete.